compiler/desugared/name_resolution.ml

```diff
@@ -235,10 +233,20 @@ let get_scope ctxt id =
         Some "Structure defined at", Mark.get (StructName.get_info sid);
       ]
       "Expecting an scope, but found a structure"
-  | exception Not_found ->
+  | exception Ident.Map.Not_found _ ->
     Message.raise_spanned_error (Mark.get id) "No scope named %s found"
       (Mark.remove id)
 
+let rec module_ctx ctxt path =
+  match path with
+  | [] -> ctxt
+  | (modname, mpos) :: path -> (
+    match ModuleName.Map.find_opt modname ctxt.modules with
+    | None ->
+      Message.raise_spanned_error mpos "Module %a not found" ModuleName.format
```

Comment:
Good point. Incidentally, I was wondering if it wouldn't be appropriate to have the various instances of `Uid.format` already define a specific color. That should work since we now discriminate between color or raw printing at the latest moment, once we know if the output is a terminal or not.